test_runner/fixtures/neon_fixtures.py

```diff
@@ -512,139 +534,80 @@ def enable_scrub_on_exit(self):
 
     def enable_remote_storage(
         self,
-        remote_storage_kind: RemoteStorageKind,
-        force_enable: bool = True,
+        kind: RemoteStorageKind,
+        force_enable: bool = False,
         enable_remote_extensions: bool = False,
     ):
-        bucket_name = re.sub(r"[_\[\]]", "-", self.test_name)[:63]
+        """
+        Configure pageserver and possibly compute extension remote storage.
 
-        if remote_storage_kind == RemoteStorageKind.NOOP:
-            return
-        elif remote_storage_kind == RemoteStorageKind.LOCAL_FS:
-            self.enable_local_fs_remote_storage(force_enable=force_enable)
-        elif remote_storage_kind == RemoteStorageKind.MOCK_S3:
-            self.enable_mock_s3_remote_storage(
-                bucket_name=bucket_name,
-                force_enable=force_enable,
-                enable_remote_extensions=enable_remote_extensions,
-            )
-        elif remote_storage_kind == RemoteStorageKind.REAL_S3:
-            self.enable_real_s3_remote_storage(
-                test_name=bucket_name,
-                force_enable=force_enable,
-                enable_remote_extensions=enable_remote_extensions,
-            )
-        else:
-            raise RuntimeError(f"Unknown storage type: {remote_storage_kind}")
+        Does not configure safekeeper remote storage.
+        """
+        assert force_enable or self.remote_storage is None, "remote storage is enabled already"
+
+        ret = self._configure_and_create_remote_storage(kind, "pageserver")
+
+        self.remote_storage = ret
+        self.remote_storage_kind = kind
+
+        if enable_remote_extensions:
+            # there is an assumption that REAL_S3 for extensions is never cleaned up
+            ext = self._configure_and_create_remote_storage(kind, "ext")
+            assert isinstance(ext, S3Storage)
+            ext.cleanup = False
+            self.ext_remote_storage = ext
+
+    def enable_safekeeper_remote_storage(self, kind: RemoteStorageKind):
```

Comment:
nit: for symmetry it would be nice to rename the other function `enable_pageserver_remote_storage`

Reply:
Yes, it would make sense. There's still the ext to decouple before that but we don't currently have those tests even enabled.

Reply:
It has now been renamed over at #5243, resolving.